In [1]:
%matplotlib inline
import pandas as pd

In [2]:
cast = pd.read_csv('cast.csv')
cast.head()

title  year      name   type                character     n
0        Closet Monster  2015  Buffy #1  actor                  Buffy 4   NaN
1       Suuri illusioni  1985    Homo $  actor                   Guests  22.0
2   Battle of the Sexes  2017   $hutter  actor          Bobby Riggs Fan  10.0
3  Secret in Their Eyes  2015   $hutter  actor          2002 Dodger Fan   NaN
4            Steve Jobs  2015   $hutter  actor  1988 Opera House Patron   NaN

### How many movies are listed in the titles dataframe?

In [3]:
cast.shape

(3611212, 6)

In [4]:
cast.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3611212 entries, 0 to 3611211
Data columns (total 6 columns):
 #   Column     Dtype  
---  ------     -----  
 0   title      object 
 1   year       int64  
 2   name       object 
 3   type       object 
 4   character  object 
 5   n          float64
dtypes: float64(1), int64(1), object(4)
memory usage: 165.3+ MB


In [5]:
cast.title.nunique()

172294

- Total 3611212 movies
- Total 172294 Unqiue movies

### What are the earliest two films listed in the titles dataframe?

In [6]:
## First need to sort by year in asc order then find top 2 movies

In [7]:
cast.sort_values('year', ascending=True)

title  year               name     type  \
2514470             Miss Jerry  1894    Blanche Bayliss  actress   
555929              Miss Jerry  1894     Chauncey Depew    actor   
464860              Miss Jerry  1894  William Courtenay    actor   
1718214  Soldiers of the Cross  1900          Reg Perry    actor   
1718063  Soldiers of the Cross  1900        Orrie Perry    actor   
...                        ...   ...                ...      ...   
2756653  The Supreme Gentleman  2023  Erika Etzelmiller  actress   
28489                Inside Me  2023     Antonio Alcala    actor   
2800939            Gnome Alone  2023           Becky G.  actress   
3592140            Fetish Trap  2023     Karen Yamileth  actress   
1697058              Inside Me  2023       Natalee Paul    actor   

                                                 character   n  
2514470               Miss Geraldine Holbrook (Miss Jerry) NaN  
555929   Himself - the Director of the New York Central... NaN  
464860                                        Mr. Hamilton NaN  
1718214                                               Lion NaN  
1718063                                               Lion NaN  
...                                                    ...  ..  
2756653                                       Hot Chick #1 NaN  
28489                                                  Max NaN  
2800939                                              Chlo? NaN  
3592140                          Attorney Allison Peterson NaN  
1697058                                               Lucy NaN  

[3611212 rows x 6 columns]

In [8]:
cast.sort_values('year', ascending=True).head(2)

title  year             name     type  \
2514470  Miss Jerry  1894  Blanche Bayliss  actress   
555929   Miss Jerry  1894   Chauncey Depew    actor   

                                                 character   n  
2514470               Miss Geraldine Holbrook (Miss Jerry) NaN  
555929   Himself - the Director of the New York Central... NaN

### How many movies have the title "Hamlet"?

In [14]:
cast[cast.title == 'Hamlet']

title  year              name     type              character     n
1615     Hamlet  1996        Riz Abbasi    actor  Attendant to Claudius   1.0
7438     Hamlet  1921  Fritz Achterberg    actor            Fortinbras,   9.0
11242    Hamlet  2009      Hayden Adams    actor                Laertes   7.0
11243    Hamlet  2009      Hayden Adams    actor                 Player   7.0
13907    Hamlet  1913       Eric Adeney    actor               Reynaldo  14.0
...         ...   ...               ...      ...                    ...   ...
3505738  Hamlet  1969    Jennifer Tudor  actress             Court lady  23.0
3528322  Hamlet  2000      Diane Venora  actress               Gertrude   3.0
3557773  Hamlet  1996     Perdita Weeks  actress          Second Player  44.0
3562331  Hamlet  2015   Katie (II) West  actress                Ophelia   7.0
3578802  Hamlet  1996      Kate Winslet  actress                Ophelia  46.0

[313 rows x 6 columns]

In [20]:
cast[cast.title == 'Hamlet'].count()

title        313
year         313
name         313
type         313
character    313
n            248
dtype: int64

- 313 movies with title Hamlet

### How many movies are titled "North by Northwest"?

In [21]:
cast[cast.title == 'North by Northwest']

title  year               name     type  \
12151    North by Northwest  1959      Stanley Adams    actor   
27852    North by Northwest  1959         Andy Albin    actor   
54093    North by Northwest  1959       Don Anderson    actor   
54307    North by Northwest  1959    Ernest Anderson    actor   
92195    North by Northwest  1959  Malcolm Atterbury    actor   
...                     ...   ...                ...      ...   
3361396  North by Northwest  1959    Eva Marie Saint  actress   
3416124  North by Northwest  1959        Doris Singh  actress   
3437248  North by Northwest  1959       Helen Spring  actress   
3567500  North by Northwest  1959      Susan Whitney  actress   
3578877  North by Northwest  1959     Paula Winslowe  actress   

                                 character    n  
12151                   Lieutenant Harding  NaN  
27852                               Farmer  NaN  
54093                               Worker  NaN  
54307     Porter on Twentieth Century Ltd.  NaN  
92195              Man at Prairie Crossing  NaN  
...                                    ...  ...  
3361396                        Eve Kendall  2.0  
3416124  First United Nations Receptionist  NaN  
3437248                             Bidder  NaN  
3567500                          Attendant  NaN  
3578877                   Woman at Auction  NaN  

[112 rows x 6 columns]

In [22]:
cast[cast.title == 'North by Northwest'].count()

title        112
year         112
name         112
type         112
character    112
n             16
dtype: int64

- 112 movies with title 'North by Northwest'

### When was the first movie titled "Hamlet" made?

In [31]:
cast[cast.title == 'Hamlet'].sort_values('year')

title  year               name     type        character     n
1545289  Hamlet  1910  Jean Mounet-Sully    actor           Hamlet   NaN
1233662  Hamlet  1911    Oscar Langkilde    actor          Horatio   6.0
2422953  Hamlet  1911   Einar Zangenberg    actor          Laertes   5.0
3015440  Hamlet  1911       Ella La Cour  actress         Gertrude   3.0
1592118  Hamlet  1911         Alwin Neu?    actor           Hamlet   1.0
...         ...   ...                ...      ...              ...   ...
3256505  Hamlet  2015       Maxine Peake  actress           Hamlet   1.0
3137189  Hamlet  2015        Jodie McNee  actress      Rosencrantz  10.0
3137190  Hamlet  2015        Jodie McNee  actress  2nd Gravedigger  10.0
2428915  Hamlet  2015   Ashley Zhangazha    actor          Laertes   6.0
2586749  Hamlet  2015  Michelle Butterly  actress         Margaret  11.0

[313 rows x 6 columns]

In [33]:
cast[cast.title == 'Hamlet'].sort_values('year').head(1)

title  year               name   type character   n
1545289  Hamlet  1910  Jean Mounet-Sully  actor    Hamlet NaN

### List all of the "Treasure Island" movies from earliest to most recent.

In [34]:
cast[cast.title == 'Treasure Island']

title  year                 name     type  \
9432     Treasure Island  1934         Robert Adair    actor   
13253    Treasure Island  1950         Chris Adcock    actor   
16246    Treasure Island  1985          Yves Afonso    actor   
42039    Treasure Island  1972   Cristino Almod?var    actor   
55071    Treasure Island  1934  John (XII) Anderson    actor   
...                  ...   ...                  ...      ...   
3415912  Treasure Island  1999        Rachel Singer  actress   
3471316  Treasure Island  1934          Jane Talent  actress   
3507587  Treasure Island  1999     Guinevere Turner  actress   
3553298  Treasure Island  1918   Eleanor Washington  actress   
3556764  Treasure Island  1973            Jane Webb  actress   

                          character     n  
9432                   Tom - Seaman   NaN  
13253                          Pike  26.0  
16246                French captain   8.0  
42039                       O'Brien  16.0  
55071    Pirate of the Spanish Main  18.0  
...                             ...   ...  
3415912                        Anna   6.0  
3471316                Streetwalker   NaN  
3507587                      Evelyn  14.0  
3553298             Prologue Player   NaN  
3556764                Mrs. Hawkins   6.0  

[190 rows x 6 columns]

In [35]:
cast[cast.title == 'Treasure Island'].sort_values('year')

title  year                  name     type  \
830114   Treasure Island  1918        Charles Gorman    actor   
840745   Treasure Island  1918             Joe Grant    actor   
909157   Treasure Island  1918          Edwin Harley    actor   
3298306  Treasure Island  1918      Violet Radcliffe  actress   
1715014  Treasure Island  1918            Lloyd Perl    actor   
...                  ...   ...                   ...      ...   
2420089  Treasure Island  1999          Caveh Zahedi    actor   
1637661  Treasure Island  1999         Nick Offerman    actor   
1375015  Treasure Island  1999           J.P. Manoux    actor   
1797264  Treasure Island  1999  Victor Raider-Wexler    actor   
307749   Treasure Island  1999          Bob Byington    actor   

                       character     n  
830114           Prologue Player   NaN  
840745   Undetermined child role   NaN  
909157           Prologue Player   NaN  
3298306         Long John Silver   3.0  
1715014                Black Dog   4.0  
...                          ...   ...  
2420089                   Harold   9.0  
1637661                   Samuel   2.0  
1375015           Officer Hughes  18.0  
1797264                 Samowitz  10.0  
307749                    Thomas  12.0  

[190 rows x 6 columns]

### How many movies were made in the year 1950?

In [37]:
cast[cast.year == 1950]

title  year              name     type  \
266               Palletoori Pilla  1950    Subba Rao A.V.    actor   
429                   Hallin Janne  1950    Kosti Aaltonen    actor   
430                   Hallin Janne  1950    Kosti Aaltonen    actor   
1481     Fortunes of Captain Blood  1950    Abdullah Abbas    actor   
1497          The Baron of Arizona  1950    Abdullah Abbas    actor   
...                            ...   ...               ...      ...   
3610440           La vida en broma  1950     Dalia ??iguez  actress   
3610445               Quinto patio  1950     Dalia ??iguez  actress   
3610681    Harukanari haha no kuni  1950     Chiharu ?tomo  actress   
3610827       Anderssonskans Kalle  1950  Ingrid ?stergren  actress   
3611137         Akdeniz korsanlari  1950   Muazzez ?lkerer  actress   

                                 character    n  
266                          Kampanna Dora  4.0  
429                                  Rosvo  NaN  
430                                Vartija  NaN  
1481                           Crew Member  NaN  
1497                                Indian  NaN  
...                                    ...  ...  
3610440                        Rebeca Mata  4.0  
3610445  Do?a Blanquita Z??iga de Vallarta  5.0  
3610681                    May Takebayashi  NaN  
3610827                Summer Camp Manager  NaN  
3611137                              Maria  2.0  

[22006 rows x 6 columns]

In [39]:
cast[cast.year == 1950].nunique()

title          907
year             1
name         10452
type             2
character    15007
n               52
dtype: int64

---------------------------------------------------------------------------------------------
In year 1950
- 22006 Movies
- 10452 unique movies

### How many movies were made in the year 1960?

In [46]:
cast[cast.year == 1960]

title  year                      name     type  \
68                     De zaak M.P.  1960             Frans 't Hoen    actor   
115             Thunder in Carolina  1960  2nd Marine Division Band    actor   
133                        Noi duri  1960              5 Asternovas    actor   
265      Sri Venkateswara Mahatmyam  1960        Subba Rao A.V. Jr.    actor   
268      Sri Venkateswara Mahatmyam  1960            Subba Rao A.V.    actor   
...                             ...   ...                       ...      ...   
3610216                     Domaren  1960         Inga-Lill ?hstr?m  actress   
3610248    Alla vi barn i Bullerbyn  1960                Kim ?sberg  actress   
3610684             Kawaita mizuumi  1960             Yachiyo ?tori  actress   
3611050                    Omringet  1960             Kari ?ksnevad  actress   
3611127                      Venner  1960              Bibba ?ygard  actress   

                                      character     n  
68                               Hollandse boer  29.0  
115                                  Themselves   NaN  
133      Police Officers disguised as Musicians  11.0  
265                                      Brahma  10.0  
268                                  Akasa Raju   5.0  
...                                         ...   ...  
3610216                                   Nurse   NaN  
3610248                                  Britta   NaN  
3610684                                  Shizue   6.0  
3611050                              Anne Aulie   2.0  
3611127                              Kontordame   NaN  

[18456 rows x 6 columns]

In [47]:
cast[cast.year == 1960].count()

title        18456
year         18456
name         18456
type         18456
character    18456
n            11910
dtype: int64

In [41]:
cast[cast.year == 1960].nunique()

title         1088
year             1
name         12248
type             2
character    14057
n               55
dtype: int64

### How many movies were made from 1950 through 1959?

In [54]:
cast[(cast.year >= 1950) & (cast.year < 1960)]

title  year                  name     type  \
67             Fire Down Below  1957  'Stretch' Cox Troupe    actor   
69          Dorp aan de rivier  1958         Frans 't Hoen    actor   
111       Ci sposeremo a Capri  1956         Quartetto 2+2    actor   
118             Die Todesarena  1953      Die 3 Motorellos    actor   
266           Palletoori Pilla  1950        Subba Rao A.V.    actor   
...                        ...   ...                   ...      ...   
3611102       Styrmand Karlsen  1958            Clara ?st?  actress   
3611103       Styrmand Karlsen  1958            Clara ?st?  actress   
3611105  Tre piger fra Jylland  1957            Clara ?st?  actress   
3611119                   Toya  1956            Aslaug ?ye  actress   
3611137     Akdeniz korsanlari  1950       Muazzez ?lkerer  actress   

                 character     n  
67          Limbo Dance by  20.0  
69                Dirk Jan  15.0  
111             Themselves  19.0  
118             Themselves   NaN  
266          Kampanna Dora   4.0  
...                    ...   ...  
3611102  Dronning Viktoria   9.0  
3611103          The Queen   9.0  
3611105          Frk. Lind   9.0  
3611119   moren til Trygve  19.0  
3611137              Maria   2.0  

[210986 rows x 6 columns]

In [58]:
cast[(cast.year >= 1950) & (cast.year < 1960)].shape

(210986, 6)

In [59]:
cast[(cast.year >= 1950) & (cast.year < 1960)].nunique()

title         10091
year             10
name          50614
type              2
character    125866
n               166
dtype: int64

-------------------------------------------------
------------------------------------------------
From 1950 to 1960
- 210986 movies
- 10091 unique movies

### In what years has a movie titled "Batman" been released?

In [81]:
cast[cast.title == 'Batman']

title  year               name     type                   character  \
95672    Batman  1943       Frank Austin    actor                 Hotel Clerk   
96186    Batman  1943     William Austin    actor           Alfred Pennyworth   
114034   Batman  1989          Del Baker    actor                 Napier Hood   
119476   Batman  1989    Michael Balfour    actor                   Scientist   
149767   Batman  1989         David Baxt    actor            Dr. Thomas Wayne   
...         ...   ...                ...      ...                         ...   
2907449  Batman  1989        Sharon Holm  actress                  Mrs. Wayne   
3253550  Batman  1943  Shirley Patterson  actress                  Linda Page   
3343947  Batman  1989           Pam Rose  actress  Party Guest at Wayne Manor   
3345872  Batman  1989          Liza Ross  actress                 Tourist Mom   
3356826  Batman  1989        Rachel Ryan  actress                      Corpse   

            n  
95672     NaN  
96186     NaN  
114034   31.0  
119476   44.0  
149767   48.0  
...       ...  
2907449  49.0  
3253550   4.0  
3343947   NaN  
3345872  45.0  
3356826   NaN  

[120 rows x 6 columns]

In [82]:
cast[cast.title == 'Batman']['year']

95672      1943
96186      1943
114034     1989
119476     1989
149767     1989
           ... 
2907449    1989
3253550    1943
3343947    1989
3345872    1989
3356826    1989
Name: year, Length: 120, dtype: int64

### How many roles were there in the movie "Inception"?

In [65]:
cast[cast.title == 'Inception']

title  year                  name     type  \
94333    Inception  2010  Michael (III) August    actor   
143138   Inception  2010          Peter Basham    actor   
176819   Inception  2010          Tom Berenger    actor   
190249   Inception  2010              Ken Bhan    actor   
250191   Inception  2010        Virgile Bramly    actor   
...            ...   ...                   ...      ...   
3238950  Inception  2010            Ellen Page  actress   
3292669  Inception  2010        Nicole Pulliam  actress   
3319073  Inception  2010    Lisa (II) Reynolds  actress   
3325768  Inception  2010         Talulah Riley  actress   
3560127  Inception  2010        Shannon Welles  actress   

                     character     n  
94333            LAX Passenger   NaN  
143138   Fischer's Jet Captain  44.0  
176819                Browning   8.0  
190249        Mombasan Gambler   NaN  
250191          Bridge Sub Con  29.0  
...                        ...   ...  
3238950                Ariadne   3.0  
3292669          Lobby Sub Con  43.0  
3319073          Private Nurse  48.0  
3325768                 Blonde  25.0  
3560127                Old Mal  51.0  

[77 rows x 6 columns]

### How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [99]:
cast[(cast.title == 'Inception')].count()

title        77
year         77
name         77
type         77
character    77
n            51
dtype: int64

In [105]:
## Movies that are ranked
cast[(cast.title == 'Inception') & (cast.n.isnull())]

title  year                     name     type  \
94333    Inception  2010     Michael (III) August    actor   
190249   Inception  2010                 Ken Bhan    actor   
361119   Inception  2010            John Ceallach    actor   
659835   Inception  2010            Seve Esposito    actor   
666157   Inception  2010      Marcus (II) Everett    actor   
682511   Inception  2010  Zachary Christopher Fay    actor   
806524   Inception  2010        Daniel Girondeaud    actor   
971196   Inception  2010          Andrew Hoagland    actor   
1056651  Inception  2010          Charles Jarrell    actor   
1069724  Inception  2010            Seong-hwan Jo    actor   
1080743  Inception  2010          Alonzo F. Jones    actor   
1146809  Inception  2010               Mobin Khan    actor   
1203175  Inception  2010            Shunsaku Kud?    actor   
1249439  Inception  2010         John (IX) Lawson    actor   
1264919  Inception  2010    Jean-Christophe Leger    actor   
1606290  Inception  2010     Fran?ois-Marie Nivon    actor   
1770685  Inception  2010             Scott Pretty    actor   
1852069  Inception  2010      Benjamin (II) Riley    actor   
1927805  Inception  2010            Norman Saleet    actor   
2050419  Inception  2010        Colin (XXV) Smith    actor   
2156986  Inception  2010             Affan Tareen    actor   
2686796  Inception  2010           Youlanda Davis  actress   
2739039  Inception  2010           Julie Eagleton  actress   
2831371  Inception  2010          Natalie Goldwin  actress   
2928772  Inception  2010       Marguerite Insolia  actress   
3016734  Inception  2010          Leticia LaBelle  actress   

                         character   n  
94333                LAX Passenger NaN  
190249            Mombasan Gambler NaN  
361119               LAX Passenger NaN  
659835             Italian Gambler NaN  
666157                Stock Broker NaN  
682511                       Extra NaN  
806524              Bridge Sub Con NaN  
971196                  Pedestrian NaN  
1056651                Hotel Guest NaN  
1069724      Japanese Landing Crew NaN  
1080743      Mombasan Casino Staff NaN  
1146809     Mombasa Casino Cashier NaN  
1203175  Saito's Bodyguard Officer NaN  
1249439              Street Person NaN  
1264919             French Student NaN  
1606290                A Passer-By NaN  
1770685                 Pedestrian NaN  
1852069             Bridge Sub Con NaN  
1927805             Airport Patron NaN  
2050419             Airport Patron NaN  
2156986                    Denison NaN  
2686796           Mombasan Cashier NaN  
2739039                Stockbroker NaN  
2831371              LAX Passenger NaN  
2928772                   Reporter NaN  
3016734             Airport Patron NaN

In [106]:
cast[(cast.title == 'Inception') & (cast.n.isnull())].count()

title        26
year         26
name         26
type         26
character    26
n             0
dtype: int64

- 26 inception are not ranked

### But how many roles in the movie "Inception" did receive an "n" value?

In [108]:
## Movies that are ranked
cast[(cast.title == 'Inception') & (~cast.n.isnull())]

title  year                  name     type  \
143138   Inception  2010          Peter Basham    actor   
176819   Inception  2010          Tom Berenger    actor   
250191   Inception  2010        Virgile Bramly    actor   
314663   Inception  2010         Michael Caine    actor   
321091   Inception  2010          Earl Cameron    actor   
417982   Inception  2010         Nicolas Clerc    actor   
428199   Inception  2010        Adam (IV) Cole    actor   
498708   Inception  2010    Jean-Michel Dagory    actor   
569504   Inception  2010     Leonardo DiCaprio    actor   
684215   Inception  2010             Russ Fega    actor   
711192   Inception  2010      Mark Fleischmann    actor   
779284   Inception  2010        Michael Gaston    actor   
783405   Inception  2010       Johnathan Geare    actor   
802478   Inception  2010         Carl Gilliard    actor   
803728   Inception  2010           Jack Gilroy    actor   
829338   Inception  2010  Joseph Gordon-Levitt    actor   
878274   Inception  2010            Lukas Haas    actor   
907468   Inception  2010             Tom Hardy    actor   
932903   Inception  2010          Ryan Hayward    actor   
1129157  Inception  2010          Tim Kelleher    actor   
1262529  Inception  2010            Tai-Li Lee    actor   
1403138  Inception  2010       Tohoru Masamune    actor   
1558174  Inception  2010        Cillian Murphy    actor   
1560914  Inception  2010     Jack (III) Murray    actor   
1609028  Inception  2010          Magnus Nolan    actor   
1640994  Inception  2010          Yuji Okumoto    actor   
1747077  Inception  2010        Andrew Pleavin    actor   
1760521  Inception  2010    Pete Postlethwaite    actor   
1794461  Inception  2010          Marc Raducci    actor   
1808640  Inception  2010            Dileep Rao    actor   
1975441  Inception  2010           Felix Scott    actor   
2168785  Inception  2010         Jason Tendell    actor   
2184854  Inception  2010        Kraig Thornber    actor   
2318555  Inception  2010          Ken Watanabe    actor   
2516169  Inception  2010      Natasha Beaumont  actress   
2658136  Inception  2010      Marion Cotillard  actress   
2668039  Inception  2010       Helena Cullinan  actress   
2697084  Inception  2010      Coralie Dedykere  actress   
2814544  Inception  2010          Claire Geare  actress   
2814552  Inception  2010          Taylor Geare  actress   
3019159  Inception  2010         Silvie Laguna  actress   
3025693  Inception  2010          Shelley Lang  actress   
3067377  Inception  2010     Alex (II) Lombard  actress   
3088316  Inception  2010         Jill Maddrell  actress   
3198324  Inception  2010      Angela Nathenson  actress   
3213496  Inception  2010         Miranda Nolan  actress   
3238950  Inception  2010            Ellen Page  actress   
3292669  Inception  2010        Nicole Pulliam  actress   
3319073  Inception  2010    Lisa (II) Reynolds  actress   
3325768  Inception  2010         Talulah Riley  actress   
3560127  Inception  2010        Shannon Welles  actress   

                       character     n  
143138     Fischer's Jet Captain  44.0  
176819                  Browning   8.0  
250191            Bridge Sub Con  29.0  
314663                     Miles  11.0  
321091          Elderly Bald Man  20.0  
417982            Bridge Sub Con  26.0  
428199               Bar Sub Con  34.0  
498708            Bridge Sub Con  30.0  
569504                      Cobb   1.0  
684215                Cab Driver  23.0  
711192           Penrose Sub Con  32.0  
779284       Immigration Officer  45.0  
783405           James (3 years)  17.0  
802478             Lobby Sub Con  40.0  
803728                  Old Cobb  50.0  
829338                    Arthur   2.0  
878274                      Nash  12.0  
907468                     Eames   4.0  
932903                    Lawyer  21.0  
1129157                 Thin Man  24.0  
1262529                  Tadashi  13.0  
1403138  Japanese Security Guard  18.0  
1

In [109]:
cast[(cast.title == 'Inception') & (~cast.n.isnull())].count()

title        51
year         51
name         51
type         51
character    51
n            51
dtype: int64

- 51 inception movies are not ranked

### Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

### Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

### Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

### How many roles were credited in the silent 1921 version of Hamlet?

### How many roles were credited in Branagh’s 1996 Hamlet?

### How many "Hamlet" roles have been listed in all film credits through history?

### How many people have played an "Ophelia"?

### How many people have played a role called "The Dude"?

### How many people have played a role called "The Stranger"?

### How many roles has Sidney Poitier played throughout his career?

### How many roles has Judi Dench played?

### List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

### List the leading roles that Cary Grant played in the 1940s in order by year.

### How many roles were available for actors in the 1950s?

### How many roles were available for actresses in the 1950s?

### How many leading roles (n=1) were available from the beginning of film history through 1980?

### How many non-leading roles were available through from the beginning of film history through 1980?

### How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?